glib/src/variant_type.rs

```diff
@@ -90,12 +90,14 @@ impl VariantType {
         VariantTy::new(&type_string)?;
 
         let len = type_string.len();
-        let ptr = type_string.into_raw();
+        unsafe {
+            let ptr = type_string.into_glib_ptr();
 
-        Ok(VariantType {
-            ptr: ptr as *mut ffi::GVariantType,
-            len,
-        })
+            Ok(VariantType {
+                ptr: ptr as *mut ffi::GVariantType,
```

Comment:
Hm why doesn't this use `ptr::NonNull`?

Reply:
Aren't you the one who wrote this code? :p 

If there is a problem with it, please an open an issue so it's not forgotten.

Reply:
I'll update it once this here is merged :) I didn't write it though